packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -72,7 +75,7 @@ class IOSDevices extends PollingDeviceDiscovery {
 
     // cancel any outstanding subscriptions.
     await _observedDeviceEventsSubscription?.cancel();
-    _observedDeviceEventsSubscription = _xcdevice.observedDeviceEvents().listen(
```

Comment:
Is the early return above not sufficient here?

Reply:
It is sufficient for this crash, but `_xcdevice.observedDeviceEvents()` is nullable, callers shouldn't need to understand why it can return null (that it can only return null if Xcode isn't installed).